brave/brave-core
# Add search query and summary from Brave Search SERP when open Leo chat

Resolves https://github.com/brave/brave-browser/issues/40615

- Add search query and summary from Brave Search SERP to conversation
  history when summarizer-key tag is present.
- Also add a new from_brave_search_SERP property in ConversationTurn to identify the
  conversation entries from Brave Search SERP.
- Modify UI to show the page context toggle when search query and summary are
  staged.
- Staged search query and summary (at the end of the conversation history) will
  be cleared if user opted out, page was unlinked, or user navigated away to a new page

For more details, please check requirement # 1 in https://docs.google.com/document/d/1idelFPpUEcKDNcyKYf3M5yw91tuIddjrlYfpaWEJjNk/edit

Security/Privacy review: https://github.com/brave/reviews/issues/1732

<img width="2555" alt="Screenshot 2024-08-22 at 2 32 58 PM" src="https://github.com/user-attachments/assets/657a0d6b-524f-4156-90b1-28caf991be10">



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Enable Brave Search `Experimental features` switch
2. Put a query in Brave Search and click `Answer with AI`
3. Wait until the answer is fully generated, open Leo panel, should see the query and summary in the panel.
4. Toggle off the page context switch should clear the staged query and summary
5. Toggle it back on should see re-staged query and summary
6. Type a follow-up question from the summary should work
